### PR TITLE
feat: add optional branch cleanup to reject command

### PR DESCRIPTION
**Pull request by prototypicalpro**
This is a simple PR to add automatic removal of branches when closing PRs with the `repo reject` command. This feature can be optionally enabled with the `--clean` flag.

### CODE REVIEW

File: src/reject-prs.ts
Comment (alexander-fenster) [inline src/reject-prs.ts:43]: This might not really fit our main use case, where branches are created by bots. But in general. is there any harm if we try to delete a branch no matter what, and just ignore the failure? (normally, it does not really matter if a branch was successfully deleted or not, as long as the PR is merged).

Comment (prototypicalpro) [reply src/reject-prs.ts:43]: I made the `--clean` flag so prevent any unexpected behavior changes, but I suppose it could also be the default behavior. In my personal use case I would always want to delete the branch. We would still need to verify the repository owner to prevent accidentally deleting a branch with the same name as a branch in a fork (e.x. if we reject a PR from `prototypicalpro/repo#feature-branch` to `googleapis/repo#master`, we don't want to accidentally delete `googleapis/repo#feature-branch`).

**Comment by JustinBeckwith** [comment]
Greetings @prototypicalpro and thanks for the PR!  I'm kind of not sure if this is a feature we should add, and I'm calling on @alexander-fenster @bcoe and @chingor13 for opinions.  Reason being - there are other more generic tools and settings that can do this exact thing.  I think? this is an option you can enable right in the GitHub repository:
https://docs.github.com/en/github/administering-a-repository/managing-the-automatic-deletion-of-branches

And if that doesn't do it, I personally use a bot to generically get this done:
https://probot.github.io/apps/delete-merged-branch/

I'm all for making this tool more useful, but I want to make sure we're doing a good job of separating concerns.

**Comment by bcoe** [comment]
@JustinBeckwith I didn't realize we could configure automated deletion like this. @prototypicalpro perhaps test and see if this would work with the `reject` workflow? I'd be open to this feature if closing doesn't perform this same cleanup step, but if it does perform the branch deletion, perhaps configuring on GitHub is the better approach.

**Comment by prototypicalpro** [comment]
[GitHub](https://docs.github.com/en/github/administering-a-repository/managing-the-automatic-deletion-of-branches) and [PRBot](https://probot.github.io/apps/delete-merged-branch/) will only automatically delete a branch if the pull request is merged, and will leave the branch alone if the PR is rejected (I assume this is to prevent work from being lost on those branches). As a result, another automation tool is needed to delete the rejected branches.

**Comment by bcoe** [comment]
@JustinBeckwith given that the auto-delete doesn't work in the reject scenario, this feature seems pretty reasonable to me.

**Comment by sofisl** [comment]
@prototypicalpro, bumping this thread! Would you mind adding some testing to this functionality? Thanks in advance!

**Comment by prototypicalpro** [comment]
Sure! I added some tests around the branch deletion functionality, is that what you were looking for?

File: test/github.ts
Comment (bcoe) [inline test/github.ts:187]:
```suggestion
```

**Review by alexander-fenster — APPROVED** [review]

**Comment by alexander-fenster** [comment]
@bcoe knows this code better than me now :)  so maybe let's wait for his approval too.

**Review by bcoe — APPROVED** [review]

**Pull request merged**
